src/ansys/aedt/core/modules/material_lib.py

```diff
@@ -850,11 +850,15 @@ def import_materials_from_file(self, input_file=None, **kwargs):
                     self.logger.warning("Material %s already exists. Renaming to %s", el, newname)
                 else:
                     newname = el
-                newmat = Material(self, newname, val, material_update=False)
-                newmat.update()
-                newmat._material_update = True
-                self.material_keys[newname] = newmat
-                materials_added.append(newmat)
+                try:
+                    newmat = Material(self, newname, val, material_update=False)
+                    newmat.update()
+                    newmat._material_update = True
+                    self.material_keys[newname] = newmat
+                    materials_added.append(newmat)
+                except KeyError as e:
+                    self.logger.error(f"Failed to import material {el!r} from {input_file!r}: key error on {e}")
+                    return False
```

Comment:
```suggestion
                    raise e
```

Reply:
Users can chose to disable the error handler and leave error being raised. In any users shoes, I'd prefer knowing that something failed by see the raised exception instead of going through logging to check if an exception occurred. Since the changes you did would break that logic, could you raise the exception back ? If there is a good reason for not doing so, please let me know. You might want to remove `{e}` from the log message if you raise `e`.

Reply:
@SMoraisAnsys thank you for pointing out the error handling setting.  Is there some reason other than historical precedent why that is enabled by default?  This often hides the true source of errors in my experience.

Re-raising the exception makes sense to me as then the user's setting for error handling will apply: if they have the catch-all turned on the method will return False and if not then they can catch the exception themselves.  I use `{e}` since for a `KeyError` it becomes the string expression for the bad key, e.g. `'Red'`.  I could be more explicit here and use `{e.args[0]}` in case this changed in a future version of Python, but for now this would allow an enterprising user to dig into the `Material` constructor and see where this key was used.

Reply:
> @SMoraisAnsys thank you for pointing out the error handling setting, I was not aware of this. Is there some reason other than historical precedent why that is enabled by default? This often hides the true source of errors in my experience.
> 
> Re-raising the exception makes sense to me as then the user's setting for error handling will apply: if they have the catch-all turned on the method will return False and if not then they can catch the exception themselves. I use `{e}` since for a `KeyError` it becomes the string expression for the bad key, e.g. `'Red'`. I could be more explicit here and use `{e.args[0]}` in case this changed in a future version of Python, but for now this would allow an enterprising user to dig into the `Material` constructor and see where this key was used.

I think it was performed to allow users to have a more resilient experience when early users worked with `pyaedt`. This might be something we would like to update when release pyaedt `v1.0`.
@Samuelopez-ansys Would this make sense or should we keep it as before ?